### ISSUE TITLE
folder password protection

### ISSUE THREAD

**Issue by zoumingzhe** [adrianschubek/dir-browser#30]
**Is your feature request related to a problem? Please describe.**
Currently, only individual files can be protected with a password.
I need to configure metadata for each file, which is quite troublesome.

**Describe the solution you'd like**
I hope individual folders can be protected with a password.
We need to traverse the metadata of all parent directories of the file and folder to ensure that password protection can overwrite all child files and folders

**Comment by zoumingzhe** [comment]
Currently, if a password is set for a folder, the key icon can be displayed, but clicking on it does not require entering a password.

**Comment by adrianschubek** [comment]
Yeah, that would be a nice feature. I would need to recursively check every parent folder's metadata if there is a password set for every request. With the current setup, this is quite inefficient/slow. Maybe a workaround would be to periodically scan all folders for a metadata file and store it in a database, and then I could simply query it. I will look into it.
